compiler/src/iree/compiler/Dialect/Flow/IR/FlowOps.cpp

```diff
@@ -151,6 +179,70 @@ static Optional<BlockArgument> getBindingArgument(Value v) {
   return llvm::None;
 }
 
+//===----------------------------------------------------------------------===//
+// custom<WorkgroupCountRegion>($body)
+//===----------------------------------------------------------------------===//
+
+static ParseResult parseWorkgroupCountRegion(OpAsmParser &parser,
+                                             Region &body) {
+  if (failed(parser.parseOptionalKeyword("workgroups"))) {
+    // Omitted.
+    return success();
+  }
+
+  SmallVector<OpAsmParser::Argument> args;
+  if (failed(parser.parseArgumentList(args, AsmParser::Delimiter::Paren,
+                                      /*allowType=*/true,
+                                      /*allowAttrs=*/true))) {
+    return failure();
+  }
+
+  // Return types must be 3 dimensions (workgroup count XYZ).
+  SmallVector<Type> returnTypes;
+  if (failed(parser.parseArrowTypeList(returnTypes))) {
+    return failure();
+  }
+  if (returnTypes.size() != 3 ||
```

Comment:
This is verifying the signature which isn't stored as part of the op and can't later be verified. That the op has `(index, index, index)` needs to be checked, otherwise one could write `(f32, tensor<4x2xf32>)` and the op would still parse.